spec/controllers/amr_heatmap_controller_spec.rb

```diff
@@ -229,64 +257,79 @@
       end
     end
 
-    describe "GET CARD entry information" do
-      it "should return relevant information from the CARD Ontology database" do
-        get :fetch_card_info, params: { geneName: "pgsA" }
-        expect(response.content_type).to eq("application/json")
-        expect(response).to have_http_status(:ok)
+    describe "GET ontology information" do
+      before do
+        s3 = Aws::S3::Client.new(stub_responses: true)
+        s3.stub_responses(:list_objects_v2, lambda { |context|
+          return {
+            is_truncated: false,
+            contents: [
+              {
+                key: "amr/ontology/#{ONTOLOGY_BUILD_DATE}/aro.json",
+                last_modified: Time.new(2019, 0o6, 20, 18, 30, 0, "-07:00"),
+                etag: "\"d70a48922d199eb2dd1ea1186c8998df\"",
+                size: 2_820_541,
+                storage_class: "STANDARD",
+              },
+            ],
+            name: context.params[:bucket].to_s,
+            prefix: context.params[:prefix].to_s,
+            max_keys: 1000,
+            key_count: 1,
+          }
+        })
+        stub_const("S3_CLIENT", s3)
+      end
+      context "when a matching gene name is found" do
+        before do
+          allow(S3Util).to receive(:s3_select_json).and_return(OQXB_ONTOLOGY.to_json + ",")
```

Comment:
Can you add a comment explaining why you need to add the comma to the end? I wouldn't know why otherwise.